Sources/Profile-Export.php

```diff
@@ -1865,7 +1865,20 @@ function ($a, $b)
 	}
 
 	// Next, we need to do for JavaScript what we just did for CSS.
-	loadJavaScriptFile('https://ajax.googleapis.com/ajax/libs/jquery/' . JQUERY_VERSION . '/jquery.min.js', array('external' => true, 'seed' => false), 'smf_jquery');
+	$jQueryUrls = array ('google_cdn' => 'https://ajax.googleapis.com/ajax/libs/jquery/'. JQUERY_VERSION . '/jquery.min.js', 'jquery_cdn' => 'https://code.jquery.com/jquery-'. JQUERY_VERSION . '.min.js', 'microsoft_cdn' => 'https://ajax.aspnetcdn.com/ajax/jQuery/jquery-'. JQUERY_VERSION . '.min.js');
+	
+	if (array_key_exists($modSettings['jquery_source'], $jQueryUrls))
+		loadJavaScriptFile($jQueryUrls[$modSettings['jquery_source']], array('external' => true, 'seed' => false), 'smf_jquery');
+
+	elseif (isset($modSettings['jquery_source']) && $modSettings['jquery_source'] == 'local')
+		loadJavaScriptFile('jquery-' . JQUERY_VERSION . '.min.js', array('seed' => false), 'smf_jquery');
+
+	elseif (isset($modSettings['jquery_source'], $modSettings['jquery_custom']) && $modSettings['jquery_source'] == 'custom')
+		loadJavaScriptFile($modSettings['jquery_custom'], array('external' => true, 'seed' => false), 'smf_jquery');
+
+	// Fall back to the local source
+	else
+		loadJavaScriptFile('jquery-' . JQUERY_VERSION . '.min.js', array('seed' => false), 'smf_jquery');
```

Comment:
Please revert these changes in Profile-Export.php.

In this case, the jQuery library *always* needs to be loaded from a reliable CDN. This is because the export files are meant to function as permanent, standalone files that will continue to work as expected even if the forum they were exported from disappears one day.

Reply:
I can't agree that Google is in anyway more reliable than the other CDN's, or even the local copy since the page is served from the same place, Google are just as fallable as the others and a user would likely have the chosen CDN's copy in cache by the time they land on this page as well but if it needs to be changed back then it shall be done, there is probably more to this one than I know.

Reply:
> ... or even the local copy since the page is served from the same place...

This seems to be the source of the misunderstanding. ProfileExport.php doesn't generate dynamic pages to be served to the browser. It generates static HTML files that the user can download to their own hard drive for archival purposes. Therefore, _everything_ in these files must be either (a) served from localhost, (b) embedded directly in the HTML file, or (c) served from a very reliable external source, where "reliable" means "highly unlikely to succumb to link rot any time in the foreseeable future"—in other words, from a major CDN. So yes, this really does need to be changed back.